Blog for Istio DNS certificate management (#5287)
istio/istio.io: content/en/blog/2019/dns-cert/index.md

---
title: Istio DNS certificate management
description: Provision and manage DNS certificates in Istio.
publishdate: 2019-11-08
attribution: Lei Tang (Google)
keywords: [security, kubernetes, certificates, DNS]
target_release: 1.4
---

By default, Citadel manages the DNS certificates of the Istio control plane.
Citadel is a large component that maintains its own private signing key, and acts as a Certificate Authority (CA).

> New in Istio 1.4, we introduce a feature to securely provision and manage DNS certificates

Reviewer: should we state it is experimental feature here?

Author: The user guide has stated it is an experimental feature.

signed by the Kubernetes CA, which has the following advantages.

* Lighter weight DNS certificate management with no dependency on Citadel.

* Unlike Citadel, this feature doesn't maintain a private signing key, which enhances security.

* Simplified root certificate distribution to TLS clients.

> Clients no longer need to wait for Citadel to generate and distribute its CA certificate.

Reviewer: what is the clients here?   sidecar proxy or istio control plane components?

Reviewer: or is this citadel agent?

Reviewer: Another thing that would be useful is to mention if I use chiron, do i still need citadel?  if so, why?

Author: The clients here are istio control plane components. If you use Chiron to provision control plane DNS certificates, you do not need Citadel to provision control plane certificates.

---
title: Istio DNS certificate management
description: Provision and manage DNS certificates in Istio.

> publishdate: 2019-10-29

Reviewer 2: Update this.

Author: Done.

attribution: Lei Tang (Google)
keywords: [security, kubernetes, certificates, DNS]
target_release: 1.4
---

By default, Citadel manages the DNS certificates of the Istio control plane.
Citadel is a large component that maintains its own private signing key, and acts as a Certificate Authority (CA).

New in Istio 1.4, we introduce a feature to securely provision and manage DNS certificates
signed by the Kubernetes CA, which has the following advantages.

> * No dependency on Citadel for DNS certificates and is

Reviewer 3: ```suggestion
* Lighter weight DNS certificate management with no dependency on Citidel.
```

Author: Revised accordingly.

> more lightweight than Citadel.

Reviewer 3: ```suggestion
```

Author: Revised accordingly.

> * Unlike Citadel, this feature doesn't require maintaining a private signing key, which enhances security.

Reviewer 3: ```suggestion
* Unlike Citadel, this implementation doesn't maintain a private signing key, which enhances security.
```

Author: Revised accordingly.

* Unlike Citadel, this feature doesn't require maintaining a private signing key, which enhances security.

* This feature simplifies root certificate distribution to TLS clients
as there is no waiting for Citadel to generate and distribute its CA cert.

The following diagram shows the architecture of provisioning and managing DNS certificates in Istio.

> {{< image width="50%"

Reviewer 4: We removed the name Chiron from the text so now the diagram needs it removed also, otherwise readers won't know what Chiron means in the diagram. Is there a way to describe this new features behavior without referring to the internal name of it?

Everything else looks great, thanks!!!

Author: The architecture needs to have concrete components (i.e., Galley, Sidecar Injector, Chiron). Otherwise, it will be too vague and difficult for readers to understand. I have added one sentence to explain the Chiron component.

    link="./architecture.png"
    caption="The architecture of provisioning and managing DNS certificates in Istio"
    >}}

> To try this new feature, please follow its [user guide](/docs/tasks/security/dns-cert).

Reviewer 3: ```suggestion
To try this new feature, refer to the [DNS certificate management task](/docs/tasks/security/dns-cert).
```

Author: Done.

---
title: Istio DNS certificate management
description: Provision and manage DNS certificates in Istio.
publishdate: 2019-10-29
attribution: Lei Tang (Google)
keywords: [security, kubernetes, certificates, DNS]
target_release: 1.4
---

> The DNS certificates used by Istio control plane components (e.g., Galley, Sidecar Injector)

Author: Revised accordingly.

> are by default managed by Citadel, which is a large component, maintains its own private

Author: Revised accordingly.

> signing key, and also acts as a CA.

Author: Revised accordingly.

> In Istio 1.4, we introduce a new feature to securely provision and manage DNS certificates

Author: Revised accordingly.

> signed by Kubernetes CA in Istio, which has the following advantages.

Author: Revised accordingly.

> * This feature removes the dependency on Citadel for DNS certificates and is

Author: Revised accordingly.

> more lightweight than Citadel.

Reviewer 5: What does "lightweight" mean? Any improved performance comes from the K8s CA doing the heavy lifting. Since it is not clear, I suggest just removing it.
```suggestion
```

Author: "lightweight" means that this feature has simpler implementation, less code, less moving parts.

> * Unlike Citadel, this feature doesn't require maintaining a private signing key, which enhances security.

Reviewer 5: ```suggestion
* No longer requires you to maintain a private signing key, which enhances security.
```

Author: The current sentence is more clearer since "you" in the suggested text is unclear (Citadel maintains a private key, not you).

> * This feature simplifies root certificate distribution to TLS clients

Reviewer 5: ```suggestion
- Simplified root certificate distribution to TLS clients. Clients
```

Author: Revised accordingly.

> as there is no waiting for Citadel to generate and distribute its CA cert.

Reviewer 5: ```suggestion
  no longer need to wait for Citadel to generate and distribute its CA certificate.
```

Author: Revised accordingly.

The following diagram shows the architecture of provisioning and managing DNS certificates in Istio.

> Chiron is the component provisioning and managing DNS certificates in Istio.

Reviewer 5: This sentence is not correct AFAIK. It is the Kubernetes CA that is provisioning and managing the certificates, Chiron is just requesting and distributing them from the Kubernetes CA, correct?
```suggestion
Chiron is the component distributing the DNS certificates within your Istio deployment.
```

Author: Kubernetes CA only signs a certificate. Chiron provisions a certificate and manages its life cycle.

---
title: Istio DNS certificate management
description: Provision and manage DNS certificates in Istio.
publishdate: 2019-10-29
attribution: Lei Tang (Google)
keywords: [security, kubernetes, certificates, DNS]
target_release: 1.4
---

The DNS certificates used by Istio control plane components (e.g., Galley, Sidecar Injector)
are by default managed by Citadel, which is a large component, maintains its own private
signing key, and also acts as a CA.

In Istio 1.4, we introduce a new feature to securely provision and manage DNS certificates

> for Istio control plane components. This feature is implemented through a component called

Author: Revised into the following to decouple the feature with control plane (the feature provisions DNS certificates signed by k8s CA but does not restrict which component uses it): 
In Istio 1.4, we introduce a new feature to securely provision and manage DNS certificates signed by Kubernetes CA in Istio.

> for Istio control plane components. This feature is implemented through a component called

Author: Done.

Chiron, which is linked with Pilot.

> * Compared with Citadel, Chiron is lightweight and removes the dependency on Citadel for control plane DNS

Author: Revised the three points as follows.
- This feature removes the dependency on Citadel for DNS certificates and is more lightweight than Citadel.
- Unlike Citadel, this feature doesn’t require maintaining a private signing key, which enhances security.
- This feature simplifies root certificate distribution to TLS clients as there is no waiting for Citadel to generate and distribute its CA cert.
